FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -572,59 +556,431 @@ The kernel will be configured as follows:
+  TEST_ASSERT_EQUAL((UBaseType_t)1, pxTaskStatus.uxCurrentPriority);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.uxBasePriority);
+  TEST_ASSERT_EQUAL(0, pxTaskStatus.usStackHighWaterMark);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *                if( listLIST_ITEM_CONTAINER( &( pxTCB->xEventListItem ) ) !=
+ * NULL )
+ *                {
+ *                   pxTaskStatus->eCurrentState = eBlocked;
+ *                }
+ *                ...
+ * @endcode
+ *
+ * Cover the case where the task is blocked.
+ */
+void test_coverage_vTaskGetInfo_blocked_task(void) {
+  TCB_t xTaskTCBs[1U] = {NULL};
+  TaskStatus_t pxTaskStatus;
+  BaseType_t xFreeStackSpace = pdFALSE;
+  eTaskState taskState = eSuspended;
+
+  /* Setup the variables and structure. */
+  vListInitialise(&xSuspendedTaskList);
+  vListInitialise(&xPendingReadyList);
+
+  xTaskTCBs[0].uxPriority = 2;
+  xTaskTCBs[0].xTaskRunState = -1;
+  xYieldPendings[0] = pdFALSE;
+  pxCurrentTCBs[0] = &xTaskTCBs[0];
+  listINSERT_END(&xSuspendedTaskList, &xTaskTCBs[0].xStateListItem);
+
+  uxTopReadyPriority = 2;
+  uxSchedulerSuspended = pdTRUE;
+
+  xTaskTCBs[0].xEventListItem.pxContainer = (struct xLIST *)1;
+  vTaskGetInfo(&xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.xTaskNumber);
+  TEST_ASSERT_EQUAL(eBlocked, pxTaskStatus.eCurrentState);
+  TEST_ASSERT_EQUAL((UBaseType_t)2, pxTaskStatus.uxCurrentPriority);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.uxBasePriority);
+  TEST_ASSERT_EQUAL(0, pxTaskStatus.usStackHighWaterMark);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *       if( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE )
+ *            {
+ *                pxTaskStatus->eCurrentState = eRunning;
+ *            }
+ *            ...
+ * @endcode
+ *
+ * Cover the case in the taskTASK_IS_RUNNING() macro where the xTaskRunState
+ * is out of bounds.
+ */
+void test_coverage_vTaskGetInfo_oob_xTaskRunState(void) {
+  TCB_t xTaskTCBs[1U] = {NULL};
+  TaskStatus_t pxTaskStatus;
+  BaseType_t xFreeStackSpace = pdFALSE;
+  eTaskState taskState = eSuspended;
+
+  xTaskTCBs[0].uxPriority = 1;
+  xTaskTCBs[0].xTaskRunState = configNUMBER_OF_CORES;
+  xYieldPendings[0] = pdFALSE;
+  pxCurrentTCBs[0] = &xTaskTCBs[0];
+
+  uxTopReadyPriority = 1;
+  uxSchedulerSuspended = pdTRUE;
+
+  vTaskGetInfo(&xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.xTaskNumber);
+  TEST_ASSERT_EQUAL(eSuspended, pxTaskStatus.eCurrentState);
+  TEST_ASSERT_EQUAL((UBaseType_t)1, pxTaskStatus.uxCurrentPriority);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.uxBasePriority);
+  TEST_ASSERT_EQUAL(0, pxTaskStatus.usStackHighWaterMark);
+}
+
+static void test_prvInitialiseStack(TCB_t *pxTCB, const uint32_t ulStackDepth) {
```

Comment:
Did this function come from someplace else? Just curious if theres a better way to get the same functionality without duplicating code.

From a style point of view, if this helper function is only going to be used in this test file i would move it to the top of the file rather than mingled within the coverage tests. 

If we think this will be reused i would recommend moving the function to smp_utest_common.c